crypto/src/memo.rs

```diff
@@ -29,6 +29,20 @@ impl Default for MemoPlaintext {
     }
 }
 
+impl TryFrom<String> for MemoPlaintext {
+    type Error = anyhow::Error;
+
+    fn try_from(input: String) -> Result<MemoPlaintext, Self::Error> {
+        if input.len() > MEMO_LEN_BYTES {
+            return Err(anyhow::anyhow!("provided memo exceeds maximum memo size"));
+        }
+        let mut mp = [0u8; MEMO_LEN_BYTES];
+        mp.copy_from_slice(input.as_bytes());
```

Comment:
In the future, we might want to have some leading-byte code or something indicating whether the memo is text or binary data, but this isn't important to worry about now.